### ISSUE TITLE
Trouble with uglify

### ISSUE THREAD

**Issue by dancancro** [DaftMonk/generator-angular-fullstack#849]
I'm getting this error when I try to build my project:

```
Running "cssmin:generated" (cssmin) task
File dist/public/app/vendor.css created: 39.94 kB → 33.97 kB
File dist/public/app/app.css created: 1.45 MB → 537.85 kB
>> No "uglify" targets found.
Warning: Task "uglify" failed. Use --force to continue.

Aborted due to warnings.
```

Building a freshly generated app works fine.  I get this:

```
Running "uglify:generated" (uglify) task
File dist/public/app/vendor.js created: 1.83 MB → 366.07 kB
File dist/public/app/app.js created: 22.59 kB → 14.63 kB
```

I can't tell where the uglify task is configured in my app or the original one.  It's not in Gruntfile.js anywhere.  I figure it's using some defaults but don't know where they're set.

**Comment by kingcody** [comment]
Hey Dan! I'm not terribly sure what excatly is the problem, but I can tell you that the `cssmin:generated` and `uglify:generated` configs are "generated" on-the-fly by the `useminPrepare` task.

Do you think you could post your `Gruntfile.js` and your `package.json` somewhere so we can get a few more clues?

**Comment by dancancro** [comment]
Thanks Cody.  Here you go http://plnkr.co/edit/0bvMyxTB2ji7OtguLmMi

For extra credit, I'd appreciate any suggestions for changing the port numbers of livereload (35729) and node-inspector (5858) for a single app.  I want to run multiple apps at the same time and whenever I try changing what I think needs to be changed, nothing happens and the app just doesn't go.

**Comment by kingcody** [comment]
I don't see anything that looks terribly out of place. Whats the output from the `useminPrepare` task? Specifically the generated uglify config.

Also, as far as changing livereload port, you can set `livereload: 1337` to a port number like so (instead of `livereload: true`) in the `watch` targets. Node-inspector is configured a little differently ([see here](https://www.npmjs.com/package/grunt-node-inspector#options)), but its pretty straight forward.

The way I normally set those port numbers is to use a grunt config string like: 
```javascript
  // Define the configuration for all the tasks
  grunt.initConfig({

    // Project settings
    yeoman: {
      ...
      livereload: 1337
    },
    watch: {
      express: {
        ...
        options: {
          livereload: '<%= yeoman.livereload %>',
          ...
        }
      }
    },
    ...
```

**Comment by dancancro** [comment]
I don't really know where to find that answer.  It says `Configuration changed for concat, uglify, cssmin
`  I don't know where it makes these changes.  

I've added the complete output of `grunt build` to the [plunkr](http://plnkr.co/edit/0bvMyxTB2ji7OtguLmMi).

**Comment by dancancro** [comment]
I figured it out.  Sort of.  I had been having trouble with the script tags being injected into index.html in a bad order.  I was getting client-side errors like `Module 'app' is not available! You either misspelled the module name or forgot...`.

So I disabled the injection, which I do not quite understand, by changing the injection markers in index.html from "build:js({.tmp,client}) app/app.js" to "buildxxx:js({.tmp,client}) app/app.js".

Undoing that, got me past the uglify error, but the bad ordering situation returned.

I still don't understand how useminPrepare configures uglify, but I now get that the error was happening because it found no targets inside of injection markers in index.html.

Now I need to figure out how to get injection of script tags to work so 
1) uglify works and 
2) the script tags are in the right order so dependencies are met.

I updated the Plunkr by adding two files:
index-manually-ordered-scripts.html  - with this one, the app works but build fails
index-auto-injected-scripts.html - with this one, the app doesn't work, but build does

Ideas?
